services/hackernews/hackernews-user-karma.service.js

```diff
@@ -0,0 +1,70 @@
+import Joi from 'joi'
+import { metric } from '../text-formatters.js'
+import { BaseJsonService, NotFound } from '../index.js'
+import { anyInteger } from '../validators.js'
+
+const schema = Joi.object({
+  id: Joi.string(),
+  karma: anyInteger,
+})
+  .allow(null)
+  .required()
+
+export default class HackerNewsUserKarma extends BaseJsonService {
+  static category = 'social'
+
+  static route = {
+    base: 'hackernews/user-karma',
+    pattern: ':userid',
+  }
+
+  static examples = [
+    {
+      title: 'HackerNews User Karma',
+      namedParams: { userid: 'pg' },
+      staticPreview: this.render({ userid: 'pg', karma: '156359' }),
+    },
+  ]
+
+  static defaultBadgeData = {
+    label: 'HackerNews User Karma',
+    namedLogo: 'ycombinator',
+  }
+
+  static render({ karma, userid }) {
+    return {
+      label: userid,
+      message: metric(karma),
```

Comment:
I think the deal with karma is it can be either positive or negative, right? (Hence validating with `anyInteger`)
If so, `metric()` currently only works properly for positive numbers. I think it probably wouldn't be hard to get `metric()` to format negatives properly too though. Relevant code and tests are at:
https://github.com/badges/shields/blob/23c0406bedfc6930735e8f5ea75dfe34faf1f290/services/text-formatters.js#L53
https://github.com/badges/shields/blob/23c0406bedfc6930735e8f5ea75dfe34faf1f290/services/text-formatters.spec.js#L40
In terms of the validator, in almost all cases I'd want to validate that a metric badge is positive (if we've got a badge that reports number of stars, number of downloads, etc and its giving us a negative value I'd want that to fail - something is probably wrong) but I think changing `isMetric` to `isPositiveMetric` is going to be a pain because it is used in _all the places_ so maybe we make a `isMetricAllowNegative` validator for the edge case where we want this (probably just this one for now). Test validators are in:
https://github.com/badges/shields/blob/8a1c69ead63c312284cec6615e7de350a101fc0f/services/test-validators.js

Reply:
Made the changes. Please have a look.